include/particles/particle_functions.tcc

```diff
@@ -0,0 +1,121 @@
+// Compute mass of particle
+namespace mpm {
+namespace particle {
+
+// Compute particle mass
+template <unsigned Tdim>
+void compute_mass(std::shared_ptr<mpm::ParticleBase<Tdim>> particle) noexcept {
+  // Check if particle volume is set and material ptr is valid
+  assert(particle->volume() != std::numeric_limits<double>::max() &&
+         particle->material() != nullptr);
+
+  // Mass = volume of particle * mass_density
+  particle->update_scalar_property(
+      mpm::properties::Scalar::MassDensity, false,
+      particle->material()->template property<double>(std::string("density")));
+
```

Comment:
Could we move this assign density function into the `assign_material`? Because for the twophase, there is no pointer for the `liquid_material` in the `twophase_particle_functions.tcc`

Reply:
Thanks, I will address it in a future PR.